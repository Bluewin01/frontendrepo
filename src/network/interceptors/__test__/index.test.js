const index = require("../index");

// @ponicode
describe("index.isHandlerEnabled", () => {
  test("isHandlerEnabled true", () => {
    let result = index.isHandlerEnabled({
      handlerEnabled: true,
      hasOwnProperty: () => false,
    });
    expect(result).toMatchSnapshot();
  });

  test("isHandlerEnabled false", () => {
    let result = index.isHandlerEnabled({
      handlerEnabled: false,
      hasOwnProperty: () => false,
    });
    expect(result).toMatchSnapshot();
  });

  test("isHandlerEnabled hasOwnProperty true", () => {
    let result = index.isHandlerEnabled({
      handlerEnabled: false,
      hasOwnProperty: () => true,
    });
    expect(result).toMatchSnapshot();
  });

  test("isHandlerEnabled true", () => {
    let result = index.isHandlerEnabled({
      handlerEnabled: true,
      hasOwnProperty: () => true,
    });
    expect(result).toMatchSnapshot();
  });

  test("isHandlerEnabled undefined", () => {
    let result = index.isHandlerEnabled(undefined);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe("index.requestHandler", () => {
  test("requestHandler 404", () => {
    let result = index.requestHandler(404);
    expect(result).toMatchSnapshot();
  });

  test("requestHandler 200", () => {
    let result = index.requestHandler(200);
    expect(result).toMatchSnapshot();
  });

  test("requestHandler 400", () => {
    let result = index.requestHandler(400);
    expect(result).toMatchSnapshot();
  });

  test("requestHandler 500", () => {
    let result = index.requestHandler(500);
    expect(result).toMatchSnapshot();
  });

  test("requestHandler", () => {
    let result = index.requestHandler(Infinity);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe("index.successHandler", () => {
  test("index.successHandler 400", () => {
    let result = index.successHandler(400);
    expect(result).toMatchSnapshot();
  });

  test("index.successHandler 200", () => {
    let result = index.successHandler(200);
    expect(result).toMatchSnapshot();
  });

  test("index.successHandler 429", () => {
    let result = index.successHandler(429);
    expect(result).toMatchSnapshot();
  });

  test("index.successHandler 500", () => {
    let result = index.successHandler(500);
    expect(result).toMatchSnapshot();
  });

  test("index.successHandler 404", () => {
    let result = index.successHandler(404);
    expect(result).toMatchSnapshot();
  });

  test("index.successHandler", () => {
    let result = index.successHandler(-Infinity);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
// describe("index.errorHandler", () => {
//   test("errorHandler", () => {
//     let result = index.errorHandler({
//       config: "services_recontextualize_front_end.gif",
//     });
//     expect(result).toMatchSnapshot();
//   });
// });
