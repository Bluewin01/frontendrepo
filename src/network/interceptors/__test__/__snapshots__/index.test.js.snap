// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.isHandlerEnabled isHandlerEnabled false 1`] = `true`;

exports[`index.isHandlerEnabled isHandlerEnabled hasOwnProperty true 1`] = `false`;

exports[`index.isHandlerEnabled isHandlerEnabled true 1`] = `true`;

exports[`index.isHandlerEnabled isHandlerEnabled true 2`] = `true`;

exports[`index.isHandlerEnabled isHandlerEnabled undefined 1`] = `true`;

exports[`index.requestHandler requestHandler 1`] = `Infinity`;

exports[`index.requestHandler requestHandler 200 1`] = `200`;

exports[`index.requestHandler requestHandler 400 1`] = `400`;

exports[`index.requestHandler requestHandler 404 1`] = `404`;

exports[`index.requestHandler requestHandler 500 1`] = `500`;

exports[`index.successHandler index.successHandler 1`] = `-Infinity`;

exports[`index.successHandler index.successHandler 200 1`] = `200`;

exports[`index.successHandler index.successHandler 400 1`] = `400`;

exports[`index.successHandler index.successHandler 404 1`] = `404`;

exports[`index.successHandler index.successHandler 429 1`] = `429`;

exports[`index.successHandler index.successHandler 500 1`] = `500`;
